superset/assets/javascripts/explorev2/components/SaveModal.js

```diff
@@ -0,0 +1,245 @@
+/* eslint camel-case: 0 */
+import React, { PropTypes } from 'react';
+const $ = window.$ = require('jquery');
+import { Modal, Alert } from 'react-bootstrap';
+import Select from 'react-select';
+import { connect } from 'react-redux';
+
+const propTypes = {
+  can_edit: PropTypes.bool,
+  onHide: PropTypes.func.isRequired,
+  actions: PropTypes.object.isRequired,
+  form_data: PropTypes.object,
+  datasource_type: PropTypes.string.isRequired,
+  user_id: PropTypes.string.isRequired,
+};
+
+class SaveModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      save_to_dashboard_id: null,
+      dashboards: [],
+      alert: null,
+    };
+  }
+
+  componentDidMount() {
+    this.prepSaveDialog();
+  }
+
+  setSaveDashboard(opt) {
+    this.setState({ save_to_dashboard_id: opt.value });
+    $('#add_to_dash_existing').prop('checked', true);
+    $('.gotodash').removeAttr('disabled');
+  }
+
+  fetchDashboards() {
+    const url = '/dashboardmodelviewasync/api/read?_flt_0_owners=' + this.props.user_id;
+    $.get(url, function (data) {
```

Comment:
looks like we're duplicating this `$.get` here as well: https://github.com/airbnb/superset/pull/1612/files#diff-cb97812fa1c826050e9d4b6d69438938R119

we should move this fetching logic to actions file.